site/content/blog/2019-10-25-instrumenting-go-code-via-ast.md

```diff
@@ -0,0 +1,315 @@
+---
+title: "Instrumenting Go via AST"
+slug: instrumenting-go-code-via-ast
+date: 2019-10-25T00:00:00-04:00
+author: Eli Yukelzon
+github: reflog
+community: eli.yukelzon
+---
+
+We've been working on integrating call tracing in the server to provide exact measurements of all API and DB calls.
+We've picked [OpenTracing](https://github.com/opentracing/opentracing-go) - a lovely open source project that allows you to setup trace reporting and enables you to support [Distributed tracing](https://opentracing.io/docs/overview/what-is-tracing/)
+
+Instrumenting your API handler in Go is very straightforward - setup connection to a collection server supporting OpenTracing (we've decided to use [Jaeger](https://www.jaegertracing.io/)) spec and wrap you code in spans.
+
+To simplify this, we've added a simple tracing module:
+```go
+package tracing
+
+import (
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-lib/metrics"
+
+	"github.com/uber/jaeger-client-go"
+	jaegercfg "github.com/uber/jaeger-client-go/config"
+	jaegerlog "github.com/uber/jaeger-client-go/log"
+	"context"
```

Comment:
I feel a little obsessive saying this, but it would be amazing if these were alphabetically ordered and grouped like we do with other imports :)